### PR TITLE
feat: Project page

### PR DESCRIPTION
**Pull request by Ansaar-Syed**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### CODE REVIEW

File: packages/web/src/pages/project/[id]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  return { ...data, createdAt: dayjs(data.createdAt), updatedAt: dayjs(data.updatedAt) };
+};
+// // > {
+// //   const { name, description, location, judgeVisits, repoUrl, ...rest } = serializedProject;
+// //   return {
+// //     ...rest,
+// //     createdAt: dayjs(createdAt),
+// //     updatedAt: dayjs(updatedAt),
+// //   };
+// });
```
Comment (SpencerKaiser) [inline packages/web/src/pages/project/[id]/index.tsx:22]:
```suggestion
```

File: packages/web/src/pages/project/[id]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    };
+
+    const { id } = router.query;
+    if (router.query && id) {
+      void fetchAndSetProjects(id as string);
+    }
+  }, [router]);
+
+  return (
+    <PageContainer pageTitle={'Project Details'} heading={'Project details'} isLoading={!project}>
```
Comment (SpencerKaiser) [inline packages/web/src/pages/project/[id]/index.tsx:48]:
```suggestion
    <PageContainer pageTitle={'Project Details'} heading={'Project Details'} isLoading={!project}>
```

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          </Text>
+        </Flex>
+
+        <Flex align="center">
+          <Text fontSize="sm" fontWeight="bold" mr={2}>
+            Repo URL:
+          </Text>
+          <Link href={repoUrl} isExternal display="flex" alignItems="center">
+            {repoUrl}
+            <FiExternalLink style={{ marginLeft: '4px' }} />
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:56]:
```suggestion
            <FiExternalLink ml={2} />
```

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          <Text fontSize="sm" fontWeight="bold" mr={2}>
+            Location:
+          </Text>
+          <Text fontSize="sm" fontWeight="bold">
+            {location}
+          </Text>
+        </Flex>
+
+        <Flex align="center">
+          <Text fontSize="sm" fontWeight="bold" mr={2}>
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:51]:
```suggestion
        <Flex align="center" gap={2}>
          <Text fontSize="sm" fontWeight="bold">
```

File: packages/web/src/components/ProjectCard/ProjectCard.tsx
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:1]: Please replace all colors with colors from our palette

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+type ProjectCardProps = {
+  project: Project;
+};
+
+export const ProjectCard: React.FC<ProjectCardProps> = ({ project }) => {
+  const { name, description, location, repoUrl } = project;
+
+  return (
+    <motion.div
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:14]: I personally don't see the value in adding a hover effect to a static card, but that's a personal opinion. If you think this looks good, we can leave it in...

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+      whileHover={{ y: -10, boxShadow: '0px 10px 20px rgba(0, 0, 0, 0.2)' }}
+      transition={{ duration: 0.3, ease: 'easeInOut' }}
+    >
+      <Box
+        borderRadius="lg"
+        overflow="hidden"
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:20]: Why hide overflow?

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        p={6}
+        background="linear-gradient(45deg, #4CAF50, #2196F3)"
+        color="white"
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:23]:
```suggestion
```

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        border="2px solid"
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:24]:
```suggestion
        borderWidth="2"
```

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        borderColor="teal.500"
+        boxShadow="lg"
+        transition="all 0.3s"
+        _hover={{
+          transform: 'scale(1.05)',
+          borderColor: 'teal.300',
+        }}
+      >
+        <Heading as="h2" size="lg" mb={2} textShadow="2px 2px 4px rgba(0, 0, 0, 0.5)">
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:33]:
```suggestion
        <Heading size="lg" mb={2} textShadow="2px 2px 4px rgba(0, 0, 0, 0.5)">
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
Did you test with a project with a long description? Make sure it looks ok

File: packages/web/src/components/ProjectCard/ProjectCard.tsx (hunk 6 above)
Comment (Ansaar-Syed) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx:20]: removed it

File: packages/web/src/components/ProjectCard/ProjectCard.tsx (hunk 5 above)
Comment (Ansaar-Syed) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx:14]: it looks, but we can always remove it later

File: packages/web/src/components/ProjectCard/ProjectCard.tsx
Comment (Ansaar-Syed) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx:1]: where can i find the palette?

File: packages/web/src/components/ProjectCard/ProjectCard.tsx (hunk 3 above)
Comment (Ansaar-Syed) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx:56]: giving this error: Type '{ ml: number; }' is not assignable to type 'IntrinsicAttributes & IconBaseProps'.
  Property 'ml' does not exist on type 'IntrinsicAttributes & IconBaseProps'.

Comment (SpencerKaiser) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx:54]: Add a box around it and add the `ml` to the box

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    <motion.div
+      whileHover={{ y: -10, boxShadow: '0px 10px 20px rgba(0, 0, 0, 0.2)' }}
+      transition={{ duration: 0.3, ease: 'easeInOut' }}
+    >
+      <Box
+        borderRadius="lg"
+        p={6}
+        background="linear-gradient(45deg, #4CAF50, #2196F3)"
+        borderWidth="2"
+        borderColor="teal.500"
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:23]: Missed colors

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]

File: packages/web/src/components/ProjectCard/ProjectCard.tsx (hunk 10 above)
Comment (Ansaar-Syed) [reply packages/web/src/components/ProjectCard/ProjectCard.tsx]: done

File: packages/web/src/components/ProjectCard/ProjectCard.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+  return (
+    <motion.div
+      whileHover={{ y: -10, boxShadow: '0px 10px 20px rgba(0, 0, 0, 0.2)' }}
+      transition={{ duration: 0.3, ease: 'easeInOut' }}
+    >
+      <Box
+        borderRadius="lg"
+        p={6}
+        background="linear-gradient(45deg, #00467F, #36495A)"
```
Comment (SpencerKaiser) [inline packages/web/src/components/ProjectCard/ProjectCard.tsx:22]: This is still using non-palette colors

**Review by SpencerKaiser — APPROVED** [review]

**Pull request merged**